File: back-end/service/chapter.service.ts
import { Chapter } from "../model/chapter";
import chapterDb from "../repository/chapter.db";

const getAllChapters = async(): Promise<Chapter[]> => {
    return await chapterDb.getAllChapters();
}

const getChapter = async(chapter_number: number): Promise<Chapter> => {
    const chapter = await chapterDb.getChapter(chapter_number);

    if (!chapter) {
        throw new Error("This chapter does not exist!")
    }

    return chapter;
}

export default {
    getAllChapters,
    getChapter
}

File: back-end-java/subverobackend/src/main/java/com/subvero/subverobackend/service/ChapterService.java
package com.subvero.subverobackend.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.subvero.subverobackend.exception.NotFoundException;
import com.subvero.subverobackend.model.Chapter;
import com.subvero.subverobackend.model.ChapterType;
import com.subvero.subverobackend.repository.ChapterRepository;

@Service
public class ChapterService {
    private final ChapterRepository chapterRepository;

    public ChapterService(ChapterRepository chapterRepository) {
        this.chapterRepository = chapterRepository;
    }

    public List<Chapter> getAllChapters(ChapterType chapterType) {
        return chapterRepository.findByChapterType(chapterType);
    }

    public Chapter getChapter(int chapterNumber, ChapterType chapterType) throws NotFoundException {
        Chapter chapter = chapterRepository.findByChapterNumberAndChapterType(chapterNumber, chapterType);

        if (chapter == null) {
            throw new NotFoundException(chapterType + "chapter " + chapterNumber + " does not exist!");
        }

        return chapter;
    }
}
